### ISSUE TITLE
add documentation for zig fmt

### ISSUE THREAD

**Issue by andrewrk** [ziglang/zig#1523]
extracted from #367 

 * how to build it / it's part of stage2 but stage2 is not done yet
 * why there are no configurable settings
 * turn zig fmt : on/ off https://github.com/ziglang/zig/commit/d21a1922eb5d76b9b0d0611eaeb42c91f83234ab
 * integration into vim
 * it supports stdin/stdout, individual files, directories, and multiple arguments. writes to stdout the files modified

**Comment by monouser7dig** [comment]
> integration into vim

would be nice to support more than vim, much like zig should not be linux first, It should not be vim first, _one could argue that vim and linux user base probably overlap significantly_ 

so zig should support probably all of 
- VSC _(which somehow everyone seems to use and probably has the best out of the box tooling, python debuggers, cpp integration etc etc so **users expect this**)_
- atom
- sublime
- vim
-emacs 

with fmt, highlighting, autocomplete(later) at least 
and maybe even refactoring support(later) as far as possible (sublime can be a bit of a crutch maybe)

ref https://github.com/ziglang/sublime-zig-language/issues/7
> golang has excellent sublime plugin which alone helps tremendously, compared to the hassle of setting up a compilation database for cpp etc.....

**rust tooling** is another benchmark by which you have to compare editor support

**Comment by Hejsil** [comment]
@monouser7dig I think what he means with "integration into vim" is to show an example of how to integrate zig fmt into a text editor. zig fmt is not designed for any editor. It's just a command line tool like any other.

**Comment by monouser7dig** [comment]
> zig fmt is not designed for any editor. It's just a command line tool like any other.

sure

I think he already has a plugin for vim if I'm not mistaken (I could though)

**Comment by andrewrk** [comment]
Yeah,  @tiehuis  put all that effort into making `zig fmt` work with vim so I just wanted to point that out. The docs section on zig fmt would mention all the editors/IDEs that have zig fmt integration available.

I do agree @monouser7dig that the fact that we currently only have integration with vim, shows a bit of bias in the user base. I hope to correct that by improving tooling on Windows, most notably with #615.

**Comment by Hejsil** [comment]
I think the vscode extension have fmt support. At least the readme implies so.

**Comment by andrewrk** [comment]
Great! I will also mention that in the zig fmt docs.

**Comment by monouser7dig** [comment]
> most notably with #615.

this will be a huge improvement for me personally at least, I'll just use whatever editor supports it, currently on sublime but vim is okay as well and if vscode is where the money is...I'll go with that.

If you have a new language, this just makes it just so much faster to get into it.

---
OT:
atom seems a bit on the dead side, there is currently xray in development and on the off chance of derailing this discussion I'd like to give a hint to Andrew because he's developing this shared audio workstation and it just happens to be the case that **xray will support shared editing**...but essentially shared anything https://github.com/atom/xray/blob/master/docs/updates/2018_09_14.md 

> As long as an application can send and receive operations and feed them into this library, it should be capable of real-time collaboration with other applications.
